app/javascript/components/events/EventTagger.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import { IEvent } from "../../models/Event";
+import { useDispatch } from "react-redux";
+import useTranslation from "../../i18n/useTranslation";
+import SearchTextInput, { SearchItem } from "../shared/SearchTextInput";
+import useLoad from "../shared/useLoad";
+import update from "immutability-helper";
+import styles from "./EventTagger.css";
+import { Link } from "react-router-dom";
+import { addTagAction } from "../../reducers/tagsReducer";
+
+interface IProps {
+  editing?: boolean;
+  event: IEvent;
+  updateEvent: (e: Partial<IEvent>) => void;
+}
+
+export default function EventTagger(props: IProps) {
+  const t = useTranslation();
+  const dispatch = useDispatch();
+  const [saveLoad] = useLoad();
+
+  const addTag = (event: IEvent, name: string) => {
+    console.log("addTag called: " + name + " and " + event);
+    console.log(" [addTag] postToApi and add to draftEvent");
+    save(name);
+  };
+
+  const setTag = (id: number, name: string) => {
+    console.log("setTag called: " + id);
+    props.updateEvent(
+      update(props.event, {
+        tag_ids: {
+          $push: [id]
+        },
+        tags: {
+          $push: [{ id: id, name: name }]
+        }
+      })
+    );
+    // TODO should only happen on a newly created tags
+    // or maybe this is fine as is
+    // dispatch to add tag to redux
+    dispatch(addTagAction(id, name));
+    // then read from redux on all subsequent reads.
+  };
+
+  const removeTag = (id: number, name: string) => {
+    console.log("removeTag called: " + id + " and " + name);
+    const new_tags = props.event.tags.filter(t => t.id != id);
+    const new_tag_ids = props.event.tag_ids.filter(i => i != id);
+    props.updateEvent(
+      update(props.event, {
+        tag_ids: { $set: new_tag_ids },
+        tags: { $set: new_tags }
+      })
+    );
+  };
+
+  const save = async (name: string) => {
+    const data = await saveLoad(duluAxios =>
+      duluAxios.post("/api/tags", {
+        tags: { tagname: name }
+      })
+    );
+    if (data) {
+      if (data.tags) {
+        props.updateEvent(
+          update(props.event, {
```

Comment:
Looks like `updateEvent` takes a PartialEvent, so I think this could be:
`props.updateEvent({tags: [...props.event.tags, data.tags[0]], tag_ids: [...props.event.tag_ids, data.tags[0].id]})`